service/fx.go

```diff
@@ -91,32 +103,34 @@ func PersistenceMaxQpsFn(
 }
 
 func GrpcServerOptionsProvider(
-	logger log.Logger,
-	rpcFactory common.RPCFactory,
-	retryableInterceptor *interceptor.RetryableInterceptor,
-	telemetryInterceptor *interceptor.TelemetryInterceptor,
-	rateLimitInterceptor *interceptor.RateLimitInterceptor,
-	tracingInterceptor telemetry.ServerTraceInterceptor,
+	params GrpcServerOptionsParams,
 ) []grpc.ServerOption {
 
-	grpcServerOptions, err := rpcFactory.GetInternodeGRPCServerOptions()
+	grpcServerOptions, err := params.RpcFactory.GetInternodeGRPCServerOptions()
 	if err != nil {
-		logger.Fatal("creating gRPC server options failed", tag.Error(err))
+		params.Logger.Fatal("creating gRPC server options failed", tag.Error(err))
 	}
 
 	return append(
 		grpcServerOptions,
-		grpc.ChainUnaryInterceptor(
-			rpc.ServiceErrorInterceptor,
-			grpc.UnaryServerInterceptor(tracingInterceptor),
-			metrics.NewServerMetricsContextInjectorInterceptor(),
-			metrics.NewServerMetricsTrailerPropagatorInterceptor(logger),
-			telemetryInterceptor.UnaryIntercept,
-			rateLimitInterceptor.Intercept,
-			retryableInterceptor.Intercept,
-		),
-		grpc.ChainStreamInterceptor(
-			telemetryInterceptor.StreamIntercept,
-		),
+		grpc.ChainUnaryInterceptor(getUnaryInterceptors(params)...),
+		grpc.ChainStreamInterceptor(params.TelemetryInterceptor.StreamIntercept),
 	)
 }
+
+func getUnaryInterceptors(params GrpcServerOptionsParams) []grpc.UnaryServerInterceptor {
+	interceptors := []grpc.UnaryServerInterceptor{
+		rpc.ServiceErrorInterceptor,
+		grpc.UnaryServerInterceptor(params.TracingInterceptor),
+		metrics.NewServerMetricsContextInjectorInterceptor(),
+		metrics.NewServerMetricsTrailerPropagatorInterceptor(params.Logger),
+		params.TelemetryInterceptor.UnaryIntercept,
+	}
+
+	interceptors = append(interceptors, params.AdditionalInterceptors...)
+
+	return append(
+		interceptors,
+		params.RateLimitInterceptor.Intercept,
+		params.RetryableInterceptor.Intercept)
```

Comment:
I think it's worth adding a test that verifies that these are applied after the additional interceptors to help future-proof this code